File: Labs/week5/C_selection_sort.py
n = int(input())
lst = [0] + list(map(int, input().split()))

for i in range(n, n - 3, -1):
    mx = lst[i]
    mxid = i
    for j in range(1, i + 1):
        if lst[j] > mx:
            mx = lst[j]
            mxid = j
    lst[i], lst[mxid] = lst[mxid], lst[i]
    print(' '.join(str(lst[k]) for k in range(1, n + 1)))

File: Labs/week5/C_selection_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> lst(n + 1);
    for (int i = 1; i <= n; i++) cin >> lst[i];

    for (int i = n; i > n - 3; i--) {
        int mx = lst[i], mxid = i;
        for (int j = 1; j <= i; j++) {
            if (lst[j] > mx) {
                mx = lst[j];
                mxid = j;
            }
        }
        swap(lst[i], lst[mxid]);
        for (int j = 1; j <= n; j++) {
            cout << lst[j] << " ";
        }
        cout << endl;
    }
    return 0;
}
